nixos/tests/matrix-conduit.nix

```diff
@@ -0,0 +1,98 @@
+import ./make-test-python.nix ({ pkgs, ... } : let
+
+
+  name = "conduit";
+
+in {
+
+
+  nodes = {
+    conduit = args: {
+      services.matrix-conduit = {
+        enable = true;
+        settings.global.server_name = name;
+        settings.global.allow_registration = true;
+        extraEnvironment.RUST_BACKTRACE = "yes";
+      };
+      services.nginx = {
+        enable = true;
+        virtualHosts.${name} = {
+          enableACME = false;
+          forceSSL = false;
+          enableSSL = false;
+
+          locations."/_matrix" = {
```

Comment:
To follow upstream (https://gitlab.com/famedly/conduit/-/blob/next/DEPLOY.md#nginx) and prevent people from copying non-working configuration at least the .well-known config should probably be added.

Reply:
According to https://spec.matrix.org/unstable/server-server-api/#resolving-server-names we *should* provide the .well-known path. But as this is a test this is not directly relevant. Also it's only needed for federation. Still we could use this place here to show a good sample configuration e.g. (I hope):
```nix
{
  services.matrix-conduit = {
    enable = true;
    settings = {
      global = {
        server_name = "example.org";
        allow_encryption = true;
        allow_federation = true;
        #allow_registration = true;
      };
    };
  };

  services.nginx = {
    enable = true;
    recommendedGzipSettings = true;
    recommendedOptimisation = true;
    recommendedProxySettings = true;
    recommendedTlsSettings = true;

    virtualHosts."example.org" = {
      enableACME = true;
      forceSSL = true;

      listen = [
        { ssl = true; port = 443; addr = "0.0.0.0"; }
        { ssl = true; port = 443; addr = "[::0]"; }
        { ssl = true; port = 8448; addr = "0.0.0.0"; }
        { ssl = true; port = 8448; addr = "[::0]"; }
      ];

      extraConfig = ''
        merge_slashes off;
      '';

      locations."= /.well-known/matrix/server".extraConfig = ''
        add_header Content-Type application/json;
        return 200 '${builtins.toJSON { "m.server" = "example.org:8448"; }}';
      '';

      locations."/_matrix" = {
        proxyPass = "http://[::1]:6167$request_uri";
        extraConfig = ''
          proxy_buffering off;
        '';
      };
    };
  };

  networking.firewall.allowedTCPPorts = [ 443 80 8448 ];
}
```

Reply:
I don't think tests are a good source of example configuration. It should be either ingrained into the module system or provided in the general documentation (manual, wiki, ...).